x/faucet/keeper/keeper.go

```diff
@@ -61,14 +61,17 @@ func (k Keeper) MintAndSend(ctx sdk.Context, msg *types.MsgMint) error {
 		return types.ErrCantWithdrawStake
 	}
 
-	// refuse mint in 24 hours
+	// Refuse mint within same 24 hour period from midnight on one day to the following day
 	a, err := sdk.AccAddressFromBech32(msg.Sender)
 	if err != nil {
 		return err
 	}
 	m := k.getMintHistory(ctx, a)
+	currentTime := time.Unix(mintTime, 0)
+	midnight := time.Date(currentTime.Year(), currentTime.Month(), currentTime.Day(), 0, 0, 0, 0, time.Local)
```

Comment:
Do we really want `time.Local` here? I guess as long as we run it on a host that is in CET that will give the intended effect, but it seems like it might be easier to reason about if we pinned to UTC.
```suggestion
	midnight := time.Date(currentTime.Year(), currentTime.Month(), currentTime.Day(), 0, 0, 0, 0, time.UTC)
```